external-crates/move/crates/move-analyzer/trace-adapter/src/runtime.ts

```diff
@@ -433,13 +433,25 @@ export class Runtime extends EventEmitter {
         // local variable array
         const frameLocalLifetimeEnds = this.trace.localLifetimeEnds.get(currentFrame.id);
         if (frameLocalLifetimeEnds) {
-            for (let i = 0; i < currentFrame.locals.length; i++) {
+            const localsLength = currentFrame.locals.length;
+            for (let i = 0; i < localsLength; i++) {
                 for (let j = 0; j < currentFrame.locals[i].length; j++) {
                     if (frameLocalLifetimeEnds[j] === instructionEvent.pc) {
                         currentFrame.locals[i][j] = undefined;
                     }
                 }
             }
+            // trim shadowed scopes that have no live variables in them
+            if (localsLength > 1) {
+                for (let i = localsLength - 1; i > 0; i--) {
+                    const existingVar = currentFrame.locals[i].find(runtimeVar => {
```

Comment:
I'd probably call this `liveVar`? it matches the comment and the intent better?

Reply:
Done!